video/out/vo_gpu_next.c

```diff
@@ -800,6 +801,33 @@ static void update_options(struct vo *vo)
     p->output_levels = cparams.levels_out;
 }
 
+static void apply_target_contrast(struct priv *p, float *min_luma, float *max_luma,
```

Comment:
If this function early returns, `max_luma` never gets initialized, which seems like an odd side effect. This code flow seems needlessly confusing to me.

Reply:
It was intentional. Unless it is needed we do not define `max_luma`. For `inf` and `auto` case it is just left at `0` for libplacebo to apply it's logic.

In case we are using `max_luma` to calc `min_luma`, then we set it to ensure consistency.

Reply:
As stated on IRC I think it would be cleaner to not touch max_luma and avoid recreating libplacebo logic by just calling into `pl_color_space_nominal_luma_ex` to get the effective peak, then setting only `min_luma` to that / desired_contrast.

Reply:
Ok, this will work fine after `pl_color_space_nominal_luma_ex` is fixed of its quirks. Although I would still set min_luma and max_luma to make sure we are requesting correct things form libplacebo. But since we use `pl_color_space_nominal_luma_ex ` it will do the same thing as libplacebo would internally.